src/main/java/state/StateFactory.java

```diff
@@ -0,0 +1,34 @@
+package state;
+
+import java.io.File;
+
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeCreatedEvent;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeSubmittedEvent;
+
+public class StateFactory {
+    public static StateAccessor getStateAccessor(File pluginDir, EiffelEvent eiffelEvent) {
```

Comment:
In State yu have set event id, and get event id, I dont see the point of adding more layers just for Accessing a state when you already have the state and you should be able to get id from that. Removing Accessors would reduce the need of adding 4 classes for each event type to 3.

Reply:
I have no idea why the accessor still exists. we renamed it. It should be completely removed :)